wgpu/examples/msaa-line/main.rs

```diff
@@ -32,6 +32,7 @@ struct Example {
     sample_count: u32,
     rebuild_bundle: bool,
     config: wgpu::SurfaceConfiguration,
+    max_sample_count: u32,
```

Comment:
I think we just need to store the flags, some devices support 4 but not 2, etc.

Reply:
I'm not sure im getting it , we get the value of `max_sample_count` from `adapter.get_texture_format_features()` so we know for a fact that it's supported

Reply:
We don't get the max sample count, we get the array of supported samples. If x4 is supported, but x2 isn't, this won't catch that case.

Reply:
I'm storing the maximum sample count in `max_sample_count` which I'm getting from the array I get from `adapter.get_texture_format_features()` , and then I'm just flipping between `1` and `max_sample_count` using the arrow keys , this is not used to figure out the supported sample count , I think you're thinking of a different use for this variable ? , if you think it's more suitable , i could rename it to `supproted_sample_counts` which will store the supported sample flags , and then use it in the arrow keys if statements to figure out the max sample count .

Reply:
Ohh, I misunderstood. The msaa example _used_ to cycle through all MSAA settings supported, but that functionality apparently changed9 was changed once 4x only was being enforced.

Reply:
Oh , thanks for the clarification !